C/services/south/ingest.cpp

```diff
@@ -676,7 +682,7 @@ void Ingest::processQueue()
 		 */
 		{
 			lock_guard<mutex> guard(m_pipelineMutex);
-			if (m_filterPipeline)
+			if (m_filterPipeline && !m_filterPipeline->isShuttingDown())
```

Comment:
&& is left to right associative but which operand will be executed first is compiler dependant, second operand might be evaluated first, in which case , it will be wrong.

Reply:
I don't think so. Can you please provide some reference?